tests/bluetooth/controller/uut/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+#
+# CMakeLists.txt file for creating of mocks library.
+#
+
+add_library(uut STATIC
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/util/mem.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/util/memq.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_tx_queue.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_conn.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_conn_upd.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_cc.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_pdu.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_local.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_phy.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_enc.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_common.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_chmu.c
+)
+
+target_include_directories(uut PUBLIC
+  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/include
```

Comment:
this line indicates that this library depends on the `mock` library created.

Should there be a `target_link_libraries(uut PUBLIC mocks)` here ?
That would ensure the include path gets inherited correctly, as well as generally making sense as I believe the sources in `uut` actually links to functions provided my the mock lib.

Reply:
agreed; that makes the CMakeLists.txt in the ctrl_feature_exchange cleaner as well

Reply:
so because you now link to the `mocks` library, then this line can be removed because it is inherited.
```suggestion
```

Reply:
this line is still here.
The `mock_ctrl/include` is an include native to the mock library, so it should be inherited.

The other includes, those with `${ZEPHYR_BASE}/subsys/bluetooth` are not native to the mock library (they are rightfully native to the `bluetooth controller` library which we are not linking directly).
Therefor the `${ZEPHYR_BASE}/subsys/bluetooth` are acceptable not to inherit, but the mock include must be inherited. The `uut` itself should not know about internals of the mock lib.